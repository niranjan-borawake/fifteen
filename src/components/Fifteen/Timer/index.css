.timer.player-2 {
  transform: rotate(180deg);
}
