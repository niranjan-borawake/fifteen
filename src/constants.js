export const FIFTEEN = 15;

export const PLAYERS = {
  ONE: 1,
  TWO: 2,
};
