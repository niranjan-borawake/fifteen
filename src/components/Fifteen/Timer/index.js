import { useEffect, useState } from 'react';
import PropTypes from 'prop-types';

import { ACTIONS } from '../reducer.js';
import { PLAYERS, FIFTEEN } from '../../../constants.js';
import useInterval from '../../../hooks/useInterval';

import './index.css';

const Timer = ({ dispatch, currentPlayer }) => {
  const [timer, setTimer] = useState(FIFTEEN);
  useEffect(() => {
    if (timer === 0) {
      dispatch({ type: ACTIONS.SET_GAME_OVER });
    }
  }, [dispatch, timer]);

  useInterval(() => {
    if (timer > 0) {
      setTimer(timer - 1);
    }
  }, 1000);

  return (
    <label className={`timer player-${currentPlayer}`}>
      {`00:${timer < 10 ? '0' + timer : timer}`}{' '}
    </label>
  );
};

Timer.propTypes = {
  currentPlayer: PropTypes.oneOf([PLAYERS.ONE, PLAYERS.TWO]),
  dispatch: PropTypes.func.isRequired,
};

export default Timer;
