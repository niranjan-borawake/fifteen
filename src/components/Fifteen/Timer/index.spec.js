import { render } from '@testing-library/react';
import Timer from '.';
import { PLAYERS } from '../../../constants';

describe('Timer', () => {
  it('should reder with default start time', () => {
    expect(
      render(<Timer currentPlayer={PLAYERS.ONE} dispatch={jest.fn} />).container
    ).toMatchSnapshot();
  });
});
