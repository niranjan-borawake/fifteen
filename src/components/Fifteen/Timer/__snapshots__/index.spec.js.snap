// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Timer should reder with default start time 1`] = `
<div>
  <label
    class="timer player-1"
    data-testid="timer-player-1"
  >
    15
  </label>
</div>
`;
